pkgs/applications/science/electronics/kicad/base.nix

```diff
@@ -41,18 +41,16 @@
 , gtk3
 }:
 
-assert ngspiceSupport -> libngspice != null;
-
+assert stdenv.lib.asserts.assertMsg (!(withOCE && stdenv.isAarch64)) "OCE fails a test on Aarch64";
 let
   versionConfig = versions.${baseName};
 
-  # oce on aarch64 fails a test
-  withOCE = oceSupport && !stdenv.isAarch64;
-  withOCC = (withOCCT && !withOCE) || (oceSupport && stdenv.isAarch64);
-
   libraries = callPackages ./libraries.nix versionConfig.libVersion;
 
+  withOCC = assert (withOCCT && !withOCE) || (withOCE && stdenv.isAarch64); true;
```

Comment:
The idea was to remove this messy assignment. What is the problem with cross enabling all of these features? I understand the lack of aarch64 support, but not this one. Maybe the expression should be simpler in favor of a failing build?

Reply:
Also the use of both `withOCC` variable and `withOCCE` is confusing for someone reading this.

Reply:
I've taken this statement out and added an assertion to ensure occ and oce are not both enabled. Kicad doesn't actually impose this requirement on us; it simply uses occ if occ and oce are both specified:

```
option( KICAD_USE_OCE
    "Build tools and plugins related to OpenCascade Community Edition (default ON)"
    ON )

option( KICAD_USE_OCC
    "Build tools and plugins related to OpenCascade Technology (overrides KICAD_USE_OCE, default OFF)"
    OFF )
```

So, if you think it's better I can take out the assertion.

Reply:
No keep it, besides the typo, everything is perfect IMO - this as well.